src/hooks/useVenueAccess.ts

```diff
@@ -22,33 +22,37 @@ export const useVenueAccess = (
     onDenyAccess && onDenyAccess();
   }, [onDenyAccess, venue]);
 
+  const logout = useCallback(() => {
+    firebase
+      .auth()
+      .signOut()
+      .finally(() => {
+        denyAccess();
+      });
+  }, [denyAccess]);
+
   useEffect(() => {
     if (!venue) return;
 
     if (venue.access) {
-      const token = getLocalStorageToken(venue.id) ?? undefined;
-      if (!token) {
+      const token = getLocalStorageToken(venue.id);
```

Comment:
Should this move out of here to its own `useMemo`? Currently, this `useEffect` will only be re-run when `venue` or `logout` changes, but we probably would also want it to run if the `token` has changed too?

(not sure how important this is, up to your judgement)